fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -8,6 +8,38 @@
 from fairlearn.metrics import DemographicParity, BoundedGroupLoss
 from fairlearn.reductions.reductions_estimator import ReductionsEstimator
 from fairlearn.reductions.grid_search import QualityMetric, GridSearchResult
+from fairlearn.reductions.moments.moment import Moment, _REDUCTION_TYPE_CLASSIFICATION
+
+class _GridGenerator:
+    """A generator of a grid of points with a bounded L1 norm.
```

Comment:
This appears to be a multi-dimensional grid? So what you're constructing is a list of points with a "Manhattan distance" to the origin of less than some given bound?

Reply:
yes. the one twist is that some coordinates are only allowed to be non-negative and some both positive and negative.

Reply:
I was guessing that was the reason for the extra Booleans :-)